### ISSUE TITLE
Trying to Add a New Token Results in Error

### ISSUE THREAD

**Issue by darkwing** [MetaMask/metamask-extension#10517]
A friend tried adding a new token and received the following error:

```
TypeError: Cannot read property 'toLowerCase' of undefined   
at chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui.js:152:566345   
at Array.some (<anonymous>)   
at r.checkExistingAddresses (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui.js:152:566327)   
at chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui.js:152:603018   
at Array.map (<anonymous>)   
at d.render (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui.js:152:602927)   
at Ya (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-libs.js:81:70734)   
at Ka (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-libs.js:81:70529)   
at El (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-libs.js:81:105704)   
at bu (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-libs.js:81:90053)   
at pu (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-libs.js:81:89978)   
at iu (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-libs.js:81:87326)   
at chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgkn
```

The error occurred immediately after hitting the Add Token screen and typing a single character.

**Comment by darkwing** [comment]
I did some live debugging with the reporter tonight, and I saw they had a token in their asset list that said “0 TKN” and the “cannot get price information” warning next to it.  I looked at the state logs and saw they had the following item in their tokens array:

```
{
    "decimals": null,
    "symbol": null
},
```

Somehow a token with no address or other information is in their token list.  When they click on that item in the list, it attempts to go to the token page, but then goes back to the asset list/home screen, so the user can’t remove it.

**Comment by darkwing** [comment]
@astarinmymind and I debugged this today and we're very close.  We believe that the trigger for this issue is likely the following line in `ui/app/ducks/swaps/swaps.js`:

```
const selectedToToken =
      getToToken(state) || fetchParams?.metaData?.destinationTokenInfo || {};
```

When we change that to:

```
const selectedToToken = {}
```

...and then try to swap, it will fail as expected.  When we then go to the asset listing, we see the following:

<img width="904" alt="TKNWTF" src="https://user-images.githubusercontent.com/46655/110037598-4374c780-7d04-11eb-9dec-be3ec90bd991.png">

Doing a project search for `TKN` finds nothing, so it's very possible that the `TKN` term is coming from an API response.  We tried abusing the `updateExchangeRates` (app/scripts/controllers/token-rates.js) function's CoinGecko API call but didn't replicate the `TKN` error.  

To complete this fix, we'll need to:

1.  Remove the `{}` fallback for `selectedToToken` -- that doesn't seem safe
2. Throw an error when `addToken` is called without an address and other valid information3
3. Write a migration to remove tokens without an address from user wallets.
